### PR TITLE
apply canonical lints

### PR DESCRIPTION
**Pull request by pq**
As per discussion yesterday, here's a delint w/ canonical lints applied.

Lints fixed include:

- `avoid_redundant_argument_values`
- `curly_braces_in_flow_control_structures`
- `prefer_iterable_whereType`
- `sort_pub_dependencies`

/cc @kenzieschmoll @jacob314

### CODE REVIEW

**Comment by pq** [comment]
Up to you all if you want to merge but this does anticipate lints that will be used in future pub scoring.

File: packages/devtools_app/lib/src/table_data.dart [hunk 1]
```diff
@@ -227,7 +227,7 @@ class TreeTableData<T extends TreeNode<T>> extends TableData<T> {
   void addColumn(ColumnData<T> column) {
     // Avoid adding multiple TreeColumnData columns.
     assert(column is! TreeColumnData<T> ||
-        _columns.where((c) => c is TreeColumnData).isEmpty);
+        _columns.whereType<TreeColumnData>().isEmpty);
```
Comment (jacob314) [inline packages/devtools_app/lib/src/table_data.dart:230]: nice! Didn't know this existed.

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/devtools_app/lib/src/table_data.dart (hunk 1 above)
Comment (pq) [reply packages/devtools_app/lib/src/table_data.dart:230]: yeah.  this is a good one for sure.

**Comment by jacob314** [comment]
Thanks for doing this!

File: packages/devtools_app/lib/src/http/_http_date.dart [hunk 2]
```diff
@@ -333,17 +333,25 @@ class HttpDate {
 
     int toInt(String s) {
       int index = 0;
-      for (; index < s.length && isDigit(s[index]); index++);
+      for (; index < s.length && isDigit(s[index]); index++) {
+        ;
```
[the remark this answers is not in the document]
Comment (pq) [reply packages/devtools_app/lib/src/http/_http_date.dart:337]: At first blush I thought that `unnecessary_statements` would catch that but it doesn't appear to. 🤔

File: packages/devtools_app/lib/src/http/_http_date.dart [hunk 3]
```diff
@@ -378,9 +386,9 @@ class HttpDate {
     }
 
     int year = toInt(yearStr);
-    if (year >= 70 && year <= 99)
+    if (year >= 70 && year <= 99) {
       year += 1900;
-    else if (year >= 0 && year <= 69) year += 2000;
+    } else if (year >= 0 && year <= 69) year += 2000;
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/http/_http_date.dart:391]: DBC: it's interesting that no lints / style guide violations appear w/ the multi-line `if` body pared with an in-line `else if` body

Comment (pq) [reply packages/devtools_app/lib/src/http/_http_date.dart:391]: good catch!  

https://github.com/dart-lang/linter/issues/2416

**Pull request merged**